ang/civiawards/directives/award.directive.js

```diff
@@ -128,76 +77,101 @@
     }
 
     /**
-     * Returns Award Types to be used in the UI
+     * Check if Save button should be disabled
      *
-     * @returns {Array} award types array in a format suitable for select 2
+     * @returns {boolean} if Save button should be disabled
      */
-    function getAwardTypeSelect2Options () {
-      return _.map(AwardType.getAll(), function (awardType) {
-        return { id: awardType.value, text: awardType.label, name: awardType.name };
-      });
+    function ifSaveButtonDisabled () {
+      return !$scope.basic_details_form.$valid || $scope.submitInProgress;
     }
+
     /**
-     * Selects a tab as active
+     * Generate Award `name` from the sent text
      *
-     * @param {string} tab tab name
+     * @param {string} text string from which name would be generated
+     * @returns {string} name of the award
      */
-    function selectTab (tab) {
-      $scope.activeTab = tab;
+    function generateAwardName (text) {
+      return text
+        .replace(/ /g, '_')
+        .replace(/[^a-zA-Z0-9_]/g, '')
+        .toLowerCase();
     }
 
     /**
-     * Enable All Award Stages
+     * Saves a New Award
      */
-    function enableAllAwardStages () {
-      _.each($scope.awardStages, function (awardStage) {
-        $scope.basicDetails.selectedAwardStages[awardStage.value] = true;
-      });
+    function saveNewAward () {
+      saveAward()
+        .then(navigateToAwardEditPage);
+    }
+
+    /**
+     * Saves an existing Award and navigates to dashboard
+     */
+    function saveAndNavigateToDashboard () {
+      saveAward()
+        .then(navigateToDashboard);
     }
 
     /**
      * Save Award
+     *
+     * @returns {Promise} promise
      */
     function saveAward () {
       $scope.submitInProgress = true;
-      saveCaseTypeBasicDetails()
+
+      return saveCaseTypeBasicDetails()
         .then(saveAdditionAwardDetails)
-        .then(function () {
-          $window.location.href = '/civicrm/case/a/?case_type_category=awards#/case?case_type_category=awards';
+        .then(function (award) {
+          return award.case_type_id;
         })
         .catch(function (error) {
-          CRM.alert(error.error_message, ts('Error'), 'error');
+          if (error.error_code === 'already exists') {
+            CRM.alert('Duplicate Award Title. Please choose and different title and try again.', ts('Error'), 'error');
```

Comment:
need to wrap the message in `ts` function. Also, shouldn't it be "duplicated" ? 🤔

Reply:
Discussed with @shitijg . Decided to use `This title is already in use. Please choose another`